### PR TITLE
pre defined classes- mathematical programs-hard-q3

### PR DESCRIPTION
**Pull request by SamamthaH**
Wrote out clause for Area Sector, awaiting double literals to finish giving clearer error messages to student

### CODE REVIEW

File: src/pre_defined_classes/mathematical_program/hard/q3/MainTest.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+    public void runMain() {
+        AreaSector.main(new String[]{});
+    }
+
+    @Test
+    public void mathTest() {
+        assertEquals(24.801128670839425, Double.parseDouble(getItemByName("math")), 0.1, "Your program's math calculations are incorrect");
+    }
+}
```
Comment (Opeyem1a) [inline src/pre_defined_classes/mathematical_program/hard/q3/MainTest.java:27]: EOF newline (this means you're missing a new line at the end of the file, this is good practice for using git version controlling and just in general tbh)

File: src/pre_defined_classes/mathematical_program/hard/q3/MainTest.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+import global.BaseTest;
+import global.variables.Clause;
+import global.variables.clauses.*;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class MainTest extends BaseTest {
+    //parsons
```
Comment (Opeyem1a) [inline src/pre_defined_classes/mathematical_program/hard/q3/MainTest.java:11]: Nitpick but this should be `// Parsons` so it's exactly consistent

**Review by Opeyem1a — CHANGES_REQUESTED** [review]

File: src/pre_defined_classes/mathematical_program/hard/q3/MainTest.java (hunk 1 above)
Comment (SamamthaH) [reply src/pre_defined_classes/mathematical_program/hard/q3/MainTest.java:27]: Do we want an extra empty line at the end of file? Also how does this relate to git version control?

File: src/pre_defined_classes/mathematical_program/hard/q3/MainTest.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+import global.BaseTest;
+import global.variables.Clause;
+import global.variables.clauses.*;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class MainTest extends BaseTest {
+    //Parsons
```
Comment (Opeyem1a) [inline src/pre_defined_classes/mathematical_program/hard/q3/MainTest.java:11]: `// Parsons`

File: src/pre_defined_classes/mathematical_program/hard/q3/MainTest.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    public Clause[] testSentence() {
+        return new Clause[]{
+                new StringLiteral("The area of the sector is: "),
+                new DoubleLiteral("math")
+        };
+    }
+
+    public void runMain() {
+        AreaSector.main(new String[]{});
```
Comment (Opeyem1a) [inline src/pre_defined_classes/mathematical_program/hard/q3/MainTest.java:20]: nitpick but I kinda like the String[0] pattern more

File: src/pre_defined_classes/mathematical_program/hard/q5/MainTest.java [hunk 5]
```diff
@@ -0,0 +1,31 @@
+package pre_defined_classes.mathematical_program.hard.q5;
```
Comment (Opeyem1a) [inline src/pre_defined_classes/mathematical_program/hard/q5/MainTest.java:1]: Oh wait why is this file here?

File: src/pre_defined_classes/mathematical_program/hard/q3/MainTest.java [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+    }
+
+    @Test
+    public void mathTest() {
+        assertEquals(24.801128670839425, Double.parseDouble(getItemByName("math")), 0.1, "Your program's math calculations are incorrect");
```
Comment (Opeyem1a) [inline src/pre_defined_classes/mathematical_program/hard/q3/MainTest.java:25]: This error message could be improved a bit. Maybe more like:
"Your calculated sector area is incorrect."

**Review by Opeyem1a — CHANGES_REQUESTED** [review]

File: src/pre_defined_classes/mathematical_program/hard/q5/MainTest.java (hunk 5 above)
Comment (SamamthaH) [reply src/pre_defined_classes/mathematical_program/hard/q5/MainTest.java:1]: I have no idea. I think something happened while I was messing around with this file while waiting to be able to use negatives and it pushed it when I flipped over to working on this program.

File: src/pre_defined_classes/mathematical_program/hard/q3/MainTest.java (hunk 6 above)
Comment (SamamthaH) [reply src/pre_defined_classes/mathematical_program/hard/q3/MainTest.java:25]: Why? They're only doing the one calculation, the output already says that its the calculated sector area

**Review by Hedgemon4 — APPROVED** [review]
Good work

File: src/pre_defined_classes/mathematical_program/hard/q3/MainTest.java [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+        };
+    }
+
+    public void runMain() {
+        AreaSector.main(new String[0]);
+    }
+
+    @Test
+    public void mathTest() {
+        assertEquals(24.801128670839425, Double.parseDouble(getItemByName("area")), 0.001, "Your calculated sector area is incorrect.");
```
Comment (aadiraju) [inline src/pre_defined_classes/mathematical_program/hard/q3/MainTest.java:25]: Do we really need all of this specificity? Wouldn't it make sense to limit it to 3 decimal values after the decimal place?
So this would be `24.801` instead and the delta will account for the variation that is smaller than that.

**Review by Opeyem1a — APPROVED** [review]
